src/mdw/java/com/lambdaherding/edi/mdw/ch04/Artists.java

```diff
@@ -0,0 +1,37 @@
+package com.lambdaherding.edi.mdw.ch04;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.lambdaherding.edi.ch04.Artist;
+
+public class Artists {
+
+  private List<Artist> artists;
+
+  public Artists(List<Artist> artists) {
+      this.artists = artists;
+  }
+
+  public Optional<Artist> getArtist(int index) {
+      if (index < 0 || index >= artists.size()) {
+          indexException(index);
```

Comment:
Yeah, that makes sense, I think when I was doing this I thought that counted as changing functionality.